### PR TITLE
Create initial setup and workflow for docusaurus

### PR DESCRIPTION
**Pull request by HebertCL**
# TL;DR

Create initial Docusaurus setup to handle static page documentation.

## Summary

This PR handles the following components:

- Docusaurus folder structure and basic page for Path static page documentation.
- GitHub Actions workflow to handle the deployment of documentation static page.

## TODO

- [x] Setup GitHub Pages to mirror `poktroll` configuration.
- [x] Setup `path.grove.city` DNS recordset for the page.
- [x] Enable pipeline to generate the documentation automatically.

## Related issues

#42

### CODE REVIEW

File: docusaurus/README.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# Website
```
Comment (Olshansk) [inline docusaurus/README.md:1]: Is this something you added or was it a default?

I'm thinking we just update it to use `make docusaurus_start` from the main Makefile. Wdyt?

**Review by Olshansk — CHANGES_REQUESTED** [review]
@HebertCL Thanks for kicking this off!

1. PTAL at admonitions. It's the best way to add notes in the future: https://docusaurus.io/docs/markdown-features/admonitions

2. I updated the main README to use ☝️ as a reference.

3. Can you add an empty `develop` and `operate` section similar to what we have in dev.poktroll.com.

It took me a little while to figure it out the first time, but just copy-pasting the pattern we have there should be enough.

**Why now?** If you set it up/configure it, it'll incentivize the rest of the team to actually add docs :)

Screenshot 2024-11-07 at 2 08 33 PM

**Comment by HebertCL** [comment]
> 3. Can you add an empty `develop` and `operate` section similar to what we have in dev.poktroll.com.

I've taken care of this and confirm that the validation is passing and building is working correctly in local. I've went ahead and request re-review.

**Review by Olshansk — APPROVED** [review]
Let's :shipit: 

Thanks @HebertCL!

**Pull request merged**